src/vs/workbench/contrib/terminal/browser/terminalEditorInput.ts

```diff
@@ -29,10 +30,15 @@ export class TerminalEditorInput extends EditorInput {
 	}
 
 	override copy(): IEditorInput {
-		const instance = this._terminalInstanceService.createInstance({}, TerminalLocation.Editor);
+		const instance = this._copyInstance || this._terminalInstanceService.createInstance({}, TerminalLocation.Editor);
+		this._copyInstance = undefined;
 		return this._instantiationService.createInstance(TerminalEditorInput, instance);
 	}
 
+	setCopyInstance(instance: ITerminalInstance) {
+		this._copyInstance = instance;
```

Comment:
I'm confused about how this is working/ not causing the issues we were seeing before (referencing the same object). But tested the code and it seems to work

Reply:
Good point, I'll document it. It's a bit weird because we rely on the `splitEditor` command and this mechanism is how we use `copy` to return specific profiles